elasticsearch/client/__init__.py

```diff
@@ -1159,7 +1160,158 @@ def msearch(self, body, index=None, doc_type=None, params=None):
+        """
+        if body in SKIP_IN_PATH:
+            raise ValueError("Empty value passed for a required argument 'body'.")
+        return self.transport.perform_request('GET', _make_path(index,
+            doc_type, '_mpercolate'), params=params, body=self._bulk_body(body),
+            headers={'content-type': 'application/x-ndjson'})
+
+    @query_params('allow_no_indices', 'expand_wildcards', 'ignore_unavailable',
+        'percolate_index', 'percolate_type', 'preference', 'routing', 'version',
+        'version_type')
+    def count_percolate(self, index, doc_type, id=None, body=None, params=None):
+        """
+        The percolator allows to register queries against an index, and then
+        send percolate requests which include a doc, and getting back the
+        queries that match on that doc out of the set of registered queries.
+        `<http://www.elastic.co/guide/en/elasticsearch/reference/current/search-percolate.html>`_
+
+        :arg index: The index of the document being count percolated.
+        :arg doc_type: The type of the document being count percolated.
+        :arg id: Substitute the document in the request body with a document
+            that is known by the specified id. On top of the id, the index and
+            type parameter will be used to retrieve the document from within the
+            cluster.
+        :arg body: The count percolator request definition using the percolate
+            DSL
+        :arg allow_no_indices: Whether to ignore if a wildcard indices
+            expression resolves into no concrete indices. (This includes `_all`
+            string or when no indices have been specified)
+        :arg expand_wildcards: Whether to expand wildcard expression to concrete
+            indices that are open, closed or both., default 'open', valid
+            choices are: 'open', 'closed', 'none', 'all'
+        :arg ignore_unavailable: Whether specified concrete indices should be
+            ignored when unavailable (missing or closed)
+        :arg percolate_index: The index to count percolate the document into.
+            Defaults to index.
+        :arg percolate_type: The type to count percolate document into. Defaults
+            to type.
+        :arg preference: Specify the node or shard the operation should be
+            performed on (default: random)
+        :arg routing: A comma-separated list of specific routing values
+        :arg version: Explicit version number for concurrency control
+        :arg version_type: Specific version type, valid choices are: 'internal',
+            'external', 'external_gte', 'force'
+        """
+        for param in (index, doc_type):
+            if param in SKIP_IN_PATH:
+                raise ValueError("Empty value passed for a required argument.")
+        return self.transport.perform_request('GET', _make_path(index,
+            doc_type, id, '_percolate', 'count'), params=params, body=body)
```

Comment:
several APIs have been removed from `master` as they have been removed in elasticsearch `6.0`. Either please remove them from the patch or base the patch on `5.x`